packages/meditrak-app/app/assessment/selectors.js

```diff
@@ -109,3 +118,80 @@ export const getQuestion = (state, questionId) => state.assessment.questions[que
 export const getAnswers = state => state.assessment.answers;
 
 export const getAnswerForQuestion = (state, questionId) => getAnswers(state)[questionId];
+
+const getArithmeticResult = (state, config) => {
+  const { formula, valueTranslation, defaultValues = {}, text = '' } = config.calculated;
+  const values = {};
+  const variables = getVariables(formula);
+  let translatedText = text;
+
+  variables.forEach(questionIdVariable => {
+    let answer = getAnswerForQuestion(state, questionIdVariable) || 0;
+
+    if (
+      valueTranslation &&
+      valueTranslation[questionIdVariable] &&
+      valueTranslation[questionIdVariable][answer] !== undefined
+    ) {
+      answer = valueTranslation[questionIdVariable][answer];
+    }
+
+    if (answer === undefined) {
+      answer = defaultValues[questionIdVariable];
+    }
+
+    values[questionIdVariable] = answer || 0;
+
+    translatedText = translatedText.replace(questionIdVariable, answer);
+  });
+
+  const calculatedResult = !isNaN(runArithmetic(formula, values))
+    ? Math.round(runArithmetic(formula, values) * 1000) / 1000
+    : 0;
+
+  translatedText = translatedText.replace('$result', calculatedResult);
+
+  return {
+    translatedText,
+    calculatedResult,
+  };
+};
+
+const getConditionResult = (state, config) => {
+  const { conditions } = config.calculated;
+  const result = Object.entries(conditions).find(([displayValue, valueConditions]) => {
+    return Object.entries(valueConditions).every(([questionId, { operator, operand }]) => {
+      const answer = getAnswerForQuestion(state, questionId);
+
+      if (answer === undefined) {
+        return false;
+      }
+
+      const checkConditionMethod = CONDITION_TYPE[operator];
+      return checkConditionMethod ? checkConditionMethod(answer, operand) : false;
+    });
+  });
+
+  let calculatedResult;
+
+  if (result) {
+    [calculatedResult] = result;
+  }
```

Comment:
I find the part where you have to extract the key a bit awkward, can we change to this:
```suggestion
  const checkAllConditionsMet = conditionsToCheck => 
    Object.entries(conditionsToCheck).every(([questionId, { operator, operand }]) => {
      const answer = getAnswerForQuestion(state, questionId);

      if (answer === undefined) {
        return false;
      }

      const checkConditionMethod = CONDITION_TYPE[operator];
      return checkConditionMethod ? checkConditionMethod(answer, operand) : false;
    };
  const calculatedResult = Object.keys(conditions).find(displayValue => checkAllConditionsMet(conditions[displayValue]));
```